rados/snapshot.go

```diff
@@ -0,0 +1,72 @@
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <stdlib.h>
+// #include <rados/librados.h>
+import "C"
+
+import "unsafe"
+
+// Snapshot represents a snapshot on the particular ioctx.
+type Snapshot struct {
+	ioctx *IOContext
+	name  string
+}
+
+// CreateSnapshot creates a pool-wide snapshot.
+//
+// Implements:
+//  int rados_ioctx_snap_create(rados_ioctx_t io, const char *snapname)
+func (ioctx *IOContext) CreateSnapshot(snapName string) (*Snapshot, error) {
+	if ioctx.ioctx == nil {
```

Comment:
JFYI: I added a validate() function in PR #330 to encapsulate some of this logic.
The only logical difference is that I reused ErrNotConnected for the error. Depending on your opinion you can either switch to that error here or we can discuss what error to use for future code.

Reply:
Thanks, I introduced ErrNoIOContext = errors.New("IOContext is missing") again similar to rbd.
ErrNotConnected is already used to check connection to the RADOS cluster. I can use validate() function, anyway #330 will go before mine. You can keep ErrNotConnected as of now in #330 , I can change the error to ErrNoIOContext in this PR if that's ok.